src/lib/core/CHIPTLV.h

```diff
@@ -52,6 +52,27 @@
 namespace chip {
 namespace TLV {
 
+struct uint24_t
+{
+    uint24_t() { memset(value, 0, 3); };
+    uint24_t(const uint64_t & v) { memcpy(value, &v, 3); };
```

Comment:
So we're relying on the non-explicit constructor here when we do our static_cast?  Better to make it explicit and have an explicit constructor call as needed.